src/realm/sync/client.cpp

```diff
@@ -1982,40 +1999,39 @@ void SessionWrapper::handle_pending_client_reset_acknowledgement()
 {
     REALM_ASSERT(!m_finalized);
 
-    auto pending_reset = _impl::client_reset::has_pending_reset(*m_db->start_frozen());
-    REALM_ASSERT(pending_reset);
-    m_sess->logger.info("Tracking pending client reset of type \"%1\" from %2", pending_reset->type,
-                        pending_reset->time);
+    std::optional<PendingReset> pending_reset;
+    {
```

Comment:
nit: it doesn't seem to need its own scope

Reply:
Since I moved the PendingResetStore back to being a set of static functions, I ended up keeping the extra scope for the read transaction.